x-pack/plugins/security_solution/server/endpoint/endpoint_app_context_services.ts

```diff
@@ -46,46 +37,6 @@ import {
   parseExperimentalConfigValue,
 } from '../../common/experimental_features';
 
-export interface MetadataService {
-  queryStrategy(
-    savedObjectsClient: SavedObjectsClientContract,
-    version?: MetadataQueryStrategyVersions
-  ): Promise<MetadataQueryStrategy>;
-}
-
-export const createMetadataService = (packageService: PackageService): MetadataService => {
-  return {
-    async queryStrategy(
-      savedObjectsClient: SavedObjectsClientContract,
-      version?: MetadataQueryStrategyVersions
-    ): Promise<MetadataQueryStrategy> {
-      if (version === MetadataQueryStrategyVersions.VERSION_1) {
-        return metadataQueryStrategyV1();
-      }
-      if (!packageService) {
-        throw new Error('package service is uninitialized');
-      }
-
-      if (version === MetadataQueryStrategyVersions.VERSION_2 || !version) {
-        const assets = await packageService.getInstalledEsAssetReferences(
-          savedObjectsClient,
-          'endpoint'
-        );
-        const expectedTransformAssets = assets.filter(
-          (ref) =>
-            ref.type === ElasticsearchAssetType.transform &&
-            ref.id.startsWith(metadataTransformPrefix)
-        );
```

Comment:
This is likely the _actual_ "isTransformEnabled" logic we want to keep around.

@kevinlog do we want to have code check for the super-old endpoint package that doesn't contain the transform, and give a proper error message to the user? Or rule that out as so-very unlikely we don't spend the mental (and request cycle) time worrying about it?

If we want to do the check, we should find a good place in the plugin lifecycle to do the check such that: this packageService call isn't done on every request, but it is recalculated, at a minimum, on package upgrades.